src/beagle-view/render/index.ts

```diff
@@ -161,21 +163,69 @@ function createRenderer({
 
     renderToScreen(view)
   }
-  
+
   function doFullRender(
     viewTree: BeagleUIElement<any>,
     anchor = '',
     mode: TreeUpdateMode = 'replaceComponent',
   ) {
     viewTree = runLifecycle(viewTree, 'beforeStart')
+
     let viewTreeWithIds = preProcess(viewTree)
     viewTreeWithIds = runLifecycle(viewTreeWithIds, 'beforeViewSnapshot')
+
     doPartialRender(viewTreeWithIds, anchor, mode)
   }
 
+  function doTemplateRender(
+    templateManager: TemplateManager,
+    anchor: string,
+    contexts: DataContext[][],
+    componentManager?: ComponentManager
+  ) {
+    if (!templateManager.default && (!templateManager.templates || templateManager.templates.length === 0)) {
+      throw new BeagleParseError('The template to render was not provided!')
```

Comment:
We avoid throwing errors while rendering, so if we fail to render a single component, we don't risk breaking the entire application.

Today, we throw an error only if the json structure expected by Beagle is mal-formed.

My suggestion would be to replace this with an error log and return.

```suggestion
     return logger.error(`Beagle can't do the template rendering at the node ${anchor} because it couldn't find any template to use. Please provide at least one template to the templateManager parameter.`)
```